jenkinsbuilder/cccp_index_reader.py

```diff
@@ -100,16 +108,93 @@ def get_projects_from_index(indexdlocation):
                                 dependson, notifyemail, desiredtag)
                         )
                     except Exception as e:
+                        print e
                         raise
     return projects
 
 
+def export_new_project_names(projects_names):
+    """
+    Exports the name of project
+    """
+    # opens the file defined at the start of this script
+    projects_names = "\n".join(projects_names)
+    try:
+        with open(projects_list, "w") as fin:
+            fin.write(projects_names)
+    except IOError as e:
+        print "Failed to export project names to file %s" % projects_list
+        print "I/O Error {0}:{1}".format(e.errno, e.strerror)
+    except:
+        print "Unexpected error:", sys.exc_info()[0]
+    else:
+        # change the mod of file so that jenkins user can edit it
+        print "Changing file permission 0777 of file %s" % projects_list
+        run_command(["chmod", "0777", projects_list])
```

Comment:
Is there no other alternative to changing the permissions to `0777`? We're giving global read, write, execute permissions which is not a good idea when this moves to prod.